mava/core_jax.py

```diff
@@ -69,3 +69,59 @@ def launch(
                 are primarily for debugging
             name : name of the system
         """
+
+
+class SystemBuilder(abc.ABC):
+    """Abstract system builder."""
+
+    @abc.abstractmethod
+    def data_server(self) -> List[Any]:
+        """Data server to store and serve transition data from and to system.
+
+        Returns:
+            System data server
+        """
+
+    @abc.abstractmethod
+    def parameter_server(self) -> Any:
+        """Parameter server to store and serve system network parameters.
+
+        Returns:
+            System parameter server
+        """
+
+    @abc.abstractmethod
+    def executor(
+        self, executor_id: str, data_server_client: Any, parameter_server_client: Any
```

Comment:
Is it not possible to have some sort of abstract data server client and parameter server client just so these methods know what methods these objects will have? 

(Very possible that this can't be done, just curious)

Reply:
Thanks Sasha, this is a good point. I think we should definitely change the types to correspond to the data and parameter client classes once they are instantiated. I will have a think about abstract versions for these as well. But I think we can do that in a separate PR.

Reply:
Would we lose any flexibility if we made `data_server_client` and `parameter_server_client` properties and then just called this like ` executor(executor_id=x)`? 

Also are we calling this function for each executor ? If so, why not call this for all executors and just return a dict with executor id and the corresponding executors?

Reply:
I just saw the comment on `executor_id="evaluator"` . Maybe a bool flag is easier if we using this to differentiate between eval and exec, otherwise someone might enter  `executor_id="executor_0"` thinking this refers to the specific executor.